src/main/java/io/cdap/plugin/aws/s3/common/S3ConnectorConfig.java

```diff
@@ -122,5 +132,32 @@ public void validate(FailureCollector collector) {
       collector.addFailure("The Access Key must be specified if authentication method is Access Credentials.", null)
         .withConfigProperty(NAME_ACCESS_KEY).withConfigProperty(NAME_AUTH_METHOD);
     }
+    collector.getOrThrowException();
```

Comment:
exceptions should be thrown at the end.

Reply:
If we throw exceptions at the end, then we'd be unnecessarily creating the STS client and making a call to the AWS API even when access ID or access Key is null

Reply:
If `accessKey` is null then we should not call `validateCredentials`.
Failure Collector is used to aggregate the exceptions and throw them at the end.

Reply:
you could just add `return` in line 130 and 133 to avoid call from reaching `validateCredentials` instead of calling `collector.getOrThrowException()`.

Because `collector.getOrThrowException()` is called by the caller which in this case would be `AbstractFileSource.configurePipeline()`.

Reply:
And we should we ensure we don't call `validateCredentials` when `accessKey` and `accessId` is macro.

Reply:
> you could just add `return` in line 130 and 133 to avoid call from reaching `validateCredentials` instead of calling `collector.getOrThrowException()`.
> 
> Because `collector.getOrThrowException()` is called by the caller which in this case would be `AbstractFileSource.configurePipeline()`.

Added the return statements for the specified lines. ad0fe83a9767b09eeea65ebc2574a5315e8408a4

> And we should we ensure we don't call validateCredentials when accessKey and accessId is macro.

Added the macro checks for `accessKey`, `accessId` and `sessionToken` ad0fe83a9767b09eeea65ebc2574a5315e8408a4